backend/main.py

```
@@ -714,159 +715,221 @@ def read_root():
+        if not contents:
+            raise HTTPException(status_code=400, detail="Uploaded file is empty")
+        
+        with open(staged_file_path, 'wb') as f:
+            f.write(contents)
+        logger.info(f"File '{filename}' for project '{project}' staged at: {staged_file_path}")
+
+    except Exception as e:
+        logger.error(f"Error staging uploaded file {filename}: {e}", exc_info=True)
+        if os.path.exists(staged_file_path):
+            try: os.unlink(staged_file_path)
+            except: pass # Best effort cleanup
```

Ruff (0.8.2)

Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

(SIM105)